### ISSUE TITLE
systemd-run: --pty option does not work if process finishes too fast

### ISSUE THREAD

**Issue by yuwata** [systemd/systemd#7531]
### Submission type

 - Bug report

### systemd version the issue has been seen with

I do not know. but recent snapshot 1ff8b31 has this.

### Used distribution

Fedora 27 x86_64

### In case of bug report: Expected behaviour you didn't see

`systemd-run --user --pty echo hoge` always shows "hoge".

### In case of bug report: Unexpected behaviour you saw

`systemd-run --user --pty echo hoge` randomly does not show "hoge".
```
$ systemd-run --user --pty echo hoge
Running as unit: run-u52.service
Press ^] three times within 1s to disconnect TTY.
hoge
$ systemd-run --user --pty echo hoge
Running as unit: run-u53.service
Press ^] three times within 1s to disconnect TTY.
$ systemd-run --user --pty echo hoge
Running as unit: run-u54.service
Press ^] three times within 1s to disconnect TTY.
hoge
```

### In case of bug report: Steps to reproduce the problem

Just run the above command many times.

**Comment by yuwata** [comment]
`systemd-run --user --pty bash -c 'sleep 1s; echo hoge'` always works.

**Comment by yuwata** [comment]
```
$ for ((i=0;i<100;i++)); do systemd-run -q -t --user echo hoge; done | grep hoge | wc -l
81
$ for ((i=0;i<100;i++)); do systemd-run -q -t --user bash -c 'sleep 1s; echo hoge'; done | grep hoge | wc -l
100
```

**Comment by poettering** [comment]
I am pretty sure this got introduced by 95f1d6bfecde60b245fae1ab0313b550201e7880.

**Comment by poettering** [comment]
I prepped a fix in #7549. Could you give this a whirl, and see if it fixes the issue for you? i think it will (as it fixes it for my locally), but it would be good to have confirmation
